From mark.einon@gmail.com  Wed Mar 11 15:54:10 2009
From: Mark Einon <mark.einon@gmail.com>
Date: Wed, 11 Mar 2009 22:51:41 +0000
Subject: Staging: rt2860: Fix remaining build warnings
To: gregkh@suse.de
Message-ID: <20090311225141.GA17034@EeeFedora10>


From: Mark Einon <mark.einon@gmail.com>

Fixed remaining four build warnings in drivers/staging/rt2860/:

drivers/staging/rt2860/common/mlme.c:900: warning: format ‘%d’ expects type ‘int’, but argument 2 has type ‘ULONG’
drivers/staging/rt2860/common/rtmp_init.c:2049: warning: ‘Value’ may be used uninitialized in this function
drivers/staging/rt2860/sta_ioctl.c:361: warning: ‘return’ with a value, in function returning void
drivers/staging/rt2860/sta_ioctl.c:2468: warning: ‘return’ with a value, in function returning void

Signed-off-by: Mark Einon <mark.einon@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rt2860/common/mlme.c      |    2 +-
 drivers/staging/rt2860/common/rtmp_init.c |    2 +-
 drivers/staging/rt2860/sta_ioctl.c        |    4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/rt2860/common/mlme.c
+++ b/drivers/staging/rt2860/common/mlme.c
@@ -897,7 +897,7 @@ VOID MlmePeriodicExec(
 		{
 			if ((pAd->StaCfg.bRadio == TRUE) && (pAd->SameRxByteCount < 700))
 			{
-				DBGPRINT(RT_DEBUG_TRACE, ("--->  SameRxByteCount = %d !!!!!!!!!!!!!!! \n", pAd->SameRxByteCount));
+				DBGPRINT(RT_DEBUG_TRACE, ("--->  SameRxByteCount = %lu !!!!!!!!!!!!!!! \n", pAd->SameRxByteCount));
 				pAd->SameRxByteCount = 700;
 				AsicResetBBP(pAd);
 			}
--- a/drivers/staging/rt2860/common/rtmp_init.c
+++ b/drivers/staging/rt2860/common/rtmp_init.c
@@ -2046,7 +2046,7 @@ VOID NICRestoreBBPValue(
 	IN PRTMP_ADAPTER pAd)
 {
 	UCHAR		index;
-	UCHAR		Value;
+	UCHAR		Value = 0;
 	ULONG		Data;
 
 	DBGPRINT(RT_DEBUG_TRACE, ("--->  NICRestoreBBPValue !!!!!!!!!!!!!!!!!!!!!!!  \n"));
--- a/drivers/staging/rt2860/sta_ioctl.c
+++ b/drivers/staging/rt2860/sta_ioctl.c
@@ -358,7 +358,7 @@ VOID RTMPAddKey(
 		if (pAd->StaCfg.bRadio == FALSE)
 		{
 			RTMP_SET_PSFLAG(pAd, fRTMP_PS_CAN_GO_SLEEP);
-			return (NDIS_STATUS_SUCCESS);
+			return;
 		}
 		DBGPRINT(RT_DEBUG_TRACE,("RTMPWPAAddKeyProc1==>\n"));
 		RTMPPCIeLinkCtrlValueRestore(pAd, RESTORE_HALT);
@@ -2465,7 +2465,7 @@ void fnSetCipherKey(
 		if (pAdapter->StaCfg.bRadio == FALSE)
 		{
 			RTMP_SET_PSFLAG(pAdapter, fRTMP_PS_CAN_GO_SLEEP);
-			return (NDIS_STATUS_SUCCESS);
+			return;
 		}
 		DBGPRINT(RT_DEBUG_TRACE,("RTMPWPAAddKeyProc1==>\n"));
 		RTMPPCIeLinkCtrlValueRestore(pAdapter, RESTORE_HALT);
